### ISSUE TITLE
Add code signing to Makefile

### ISSUE THREAD

**Issue by splitt3r** [nextcloud/appstore#529]
It would be very nice to have appstore code signing added to the app-templates Makefile. Started it here: https://github.com/nextcloud/polls/blob/master/Makefile.

**Comment by BernhardPosselt** [comment]
IMHO code signing should be avoided in the old form, here's why:

* once an app is installed that contains a signature.json a user can never install an app again without providing a signature.json
* signature.json is redundant since all packages downloaded from the store already contain a signature (one over the whole archive) so there is no additional security benefit
* the store does not validate that your signature.json is actually correct which leads to users not being able to upgrade their apps in certain cases see https://github.com/nextcloud/calendar/issues/611

The only benefit that you are getting is that if a user manually uploads files the signing process can detect missing files but since most apps are installed over the app store I don't really see this as a pressing use case.

TL;DR: older apps need to use code signing but we shouldnt promote it for newer apps.

**Comment by splitt3r** [comment]
Thanks for the info 👍 that makes sense.
